specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/codeVersion.json

```diff
@@ -0,0 +1,400 @@
+          },
+          {
+            "$ref": "#/parameters/CodeVersionVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/CodeVersionResource"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "machineLearningServices.json#/definitions/MachineLearningServiceError"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "CodeVersion"
+        ],
+        "description": "Create or update code.",
+        "operationId": "CodeVersions_CreateOrUpdate",
+        "x-ms-examples": {
+          "Create or Update CodeVersion": {
+            "$ref": "./examples/createOrUpdateCodeVersion.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "machineLearningServices.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/APIVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/CodeVersionNameParameter"
+          },
+          {
+            "$ref": "#/parameters/CodeVersionVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Code entity to create or update.",
+            "schema": {
+              "$ref": "#/definitions/CodeVersionResourceRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/CodeVersionResource"
+            }
+          },
+          "201": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/CodeVersionResource"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "machineLearningServices.json#/definitions/MachineLearningServiceError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "CodeVersionNameParameter": {
+      "in": "path",
+      "name": "name",
+      "type": "string",
+      "description": "Name for CodeVersion.",
+      "required": true,
+      "x-ms-parameter-location": "method"
+    },
+    "CodeVersionVersionParameter": {
+      "in": "path",
+      "name": "version",
+      "type": "string",
+      "description": "Version for CodeVersion.",
+      "required": true,
+      "x-ms-parameter-location": "method"
+    },
+    "CodeVersionVersionOptionalParameter": {
+      "in": "query",
+      "name": "version",
```

Comment:
what is this query parameter used for?  If the user wants a specific version they can do an individual GET on that version

Reply:
you're right, that shouldn't exist
please reflect changes in controller @j-so